### PR TITLE
Add built-in support for ccache

### PR DESCRIPTION
**Pull request by rongou**
This adds support for `ccache` out of the box without asking the developer to do anything on their end, at the same time still allows the flexibility to configure their own setup (e.g. using both `ccache` and `distcc`).
```console
$ ccache -C  # clear the cache
$ time make -j12
...
real	2m0.800s
user	14m59.645s
sys	0m40.054s

$ make clean  # now everything is cached
$ time make -j12
...
real	0m1.069s
user	0m6.176s
sys	0m1.868s
```

### CODE REVIEW

**Comment by harrism** [comment]
I would like @trxcllnt to comment.

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ project(
   VERSION 0.17.0
   LANGUAGES CXX)
 
+include(cmake/Modules/ccache.cmake)
```
Comment (trxcllnt) [inline CMakeLists.txt:21]: Can this be guarded by a CMake option? Changing the CMake compiler launcher will break anyone using ccache compiler rsymlinks.

**Review by trxcllnt — CHANGES_REQUESTED** [review]

**Review by mike-wendt — CHANGES_REQUESTED** [review]
Same concerns as rapidsai/cudf#7000; let's discuss this before moving forward

File: CMakeLists.txt (hunk 1 above)
Comment (rongou) [reply CMakeLists.txt:21]: Done.

**Comment by rongou** [comment]
@mike-wendt @trxcllnt updated according to offline discussion. PTAL

**Comment by rongou** [comment]
Hmm, I think I might have a fix for the nvcc symlink issue. Hold on.

**Comment by rongou** [comment]
Closing this for now. Will rely on https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5620.

**Pull request closed without merging**